sdk/core/corehttp/tests/perf_tests/_async_iterator_random_stream.py

```diff
@@ -0,0 +1,71 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+from devtools_testutils.perfstress_tests._random_stream import get_random_bytes, _DEFAULT_LENGTH
+
+
+# The AsyncIteratorRandomStream is used specifically for the AsyncHttpXTransport upload stream scenario, since the
+# async httpx transport requires the request body stream to be type AsyncIterator (i.e. have an __aiter__ and __anext__ method rather than __iter__).
+# Specific check in httpx here: https://github.com/encode/httpx/blob/7df47ce4d93a06f2c3310cd692b4c2336d7663ba/httpx/_content.py#L116.
+
+class AsyncIteratorRandomStream():
+    def __init__(self, length, initial_buffer_length=_DEFAULT_LENGTH):
+        super().__init__()
+        self._base_data = get_random_bytes(initial_buffer_length)
+        self._data_length = length
+        self._base_buffer_length = initial_buffer_length
+        self._position = 0
+        self._remaining = length
+        self._closed = False
+
+    def __len__(self):
+        return self._remaining
+    
+    def __aiter__(self):
+        return self
+    
+    async def __anext__(self):
+        if self._remaining == 0:
+            raise StopAsyncIteration
+        return self.read()
+
+    def reset(self):
+        self._position = 0
+        self._remaining = self._data_length
+        self._closed = False
+
+    def read(self, size=None):
+        if self._remaining == 0:
+            return b""
+
+        if size is None:
+            e = self._base_buffer_length
+        else:
+            e = size
+        e = min(e, self._remaining)
+        if e > self._base_buffer_length:
+            self._base_data = get_random_bytes(e)
+            self._base_buffer_length = e
+        self._remaining = self._remaining - e
+        self._position += e
+        return self._base_data[:e]
+
+    def seek(self, index, whence=0):
```

Comment:
Methods like `seek`, `tell`, and `close` are more for IO objects, and I don't think these are used. We can just delete them to keep this leaner.